src/slot_migrate.cc

```diff
@@ -104,10 +104,9 @@ Status SlotMigrate::MigrateStart(Server *svr, const std::string &node_id, const
   dst_node_ = node_id;
 
   // Create migration job
-  SlotMigrateJob *job = new SlotMigrateJob(slot, dst_ip, dst_port,
-                                           speed, pipeline_size, seq_gap);
+  slot_job_ = std::unique_ptr<SlotMigrateJob>(new SlotMigrateJob(slot, dst_ip, dst_port,
+                                           speed, pipeline_size, seq_gap));
   std::lock_guard<std::mutex> guard(job_mutex_);
-  slot_job_ = job;
```

Comment:
Does removing this line make the expected behavior different? This line is in a scope of a `lock_guard`.

Reply:
sorry, my bad. could i just move Line107 to the scope of the 'lock_guard', or maybe i need to alloc memory for a temp unique_ptr and move the temp unique_ptr's value to slot_job_ in the scope of 'lock_guard'

Reply:
Never mind, I think the latter is better.